pkgs/development/python-modules/pylama/default.nix

```diff
@@ -1,34 +1,66 @@
-{ lib, buildPythonPackage, fetchPypi
-, eradicate, mccabe, pycodestyle, pydocstyle, pyflakes
-, pytest, ipdb }:
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, substituteAll
+, git
+, eradicate
+, mccabe
+, mypy
+, pycodestyle
+, pydocstyle
+, pyflakes
+, vulture
+, pytestCheckHook
+}:
 
 buildPythonPackage rec {
   pname = "pylama";
-  version = "7.7.1";
+  version = "8.0.6";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "9bae53ef9c1a431371d6a8dca406816a60d547147b60a4934721898f553b7d8f";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "klen";
+    repo = "pylama";
+    rev = version;
+    hash = "sha256-Olq/CZ/t1wqACoknAKsvdDKnyLZkxRtHokpu33I3trg=";
   };
 
+  patches = [
+    (substituteAll {
+      src = ./paths.patch;
+      git = "${lib.getBin git}/bin/git";
+    })
+  ];
+
   propagatedBuildInputs = [
     eradicate
     mccabe
+    mypy
     pycodestyle
     pydocstyle
     pyflakes
+    vulture
   ];
 
-  checkInputs = [ pytest ipdb ];
+  checkInputs = [
+    pytestCheckHook
+  ];
 
-  # tries to mess with the file system
-  doCheck = false;
+  disabledTests = [
+    "test_pylint" # infinite recursion
+    "test_quotes" # FIXME package pylama-quotes
+    "test_radon" # FIXME package radon
+  ];
```

Comment:
These comments aren't clear to me. Why is `test_pylint` failing with an infinite recursion? Is this an issue upstream? If so, we should link to it.

What are these FIXMEs for exactly?

Reply:
pylint -> isort -> pylama

The FIXMEs are extra requires.

Reply:
We could move the tests to passthru.tests to break the infinite recursion. It is always good to test package which cause mass rebuilds indepth.

Reply:
We are still testing the package, just not the pylint component.
Also after this PR way fewer packages should depend on pylama.